Mathlib/LinearAlgebra/Matrix/Polynomial.lean

```diff
@@ -35,27 +35,29 @@ open Polynomial Matrix Equiv.Perm
 
 namespace Polynomial
 
+set_option maxHeartbeats 65000 in
```

Comment:
I think the type ascriptions in this are not actually necessary. My guess is that since `binop2.lean` is interpreted code instead of compiled code it's consuming a whole lot of heartbeats. The type ascriptions help reduce the number of heartbeats.